Frameworks/Foundation/NSArray.mm

```diff
@@ -665,6 +665,10 @@ - (NSUInteger)indexOfObject:(NSObject*)obj
                                             _CFComparatorFunctionFromComparator,
                                             &comparator);
 
+	if (index >= [self count] && (options & NSBinarySearchingInsertionIndex)) {
+		return index;
```

Comment:
nit: please run clang-format with the .clang-format file in the root of our repo; we're a 4-space-indent codebase! :)